server/src/main/java/thesistrack/ls1/repository/UserRepository.java

```diff
@@ -0,0 +1,27 @@
+package thesistrack.ls1.repository;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+import thesistrack.ls1.entity.User;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, UUID> {
+    Optional<User> findByUniversityId(String universityId);
+
+    @Query("SELECT u FROM User u JOIN UserGroup g ON (u.id = g.id.userId) WHERE " +
+            "g.id.group IN (:groups) AND " +
+            "(LOWER(u.firstName) LIKE %:searchQuery% OR " +
+            "LOWER(u.lastName) LIKE %:searchQuery% OR " +
+            "LOWER(u.email) LIKE %:searchQuery% OR " +
+            "LOWER(u.matriculationNumber) LIKE %:searchQuery% OR " +
+            "LOWER(u.universityId) LIKE %:searchQuery%)")
+    Page<User> searchUsers(@Param("searchQuery") String searchQuery, @Param("groups") Set<String> groups, Pageable page);
```

Comment:
In `ApplicationRepository.searchApplications()` you require the `groups` parameter as first param, here you require it as second. This results in switched arguments and reduces consistency. Please fix either of those 👍

Reply:
Yes makes sense